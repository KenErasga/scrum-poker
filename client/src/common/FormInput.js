import React from 'react';
import { FormControl, InputLabel, Input} from '@material-ui/core'

const FormInput = props => {
    return (
        <FormControl margin="normal" fullWidth>
            <InputLabel htmlFor="component-simple">{`${props.InputLabel}`}</InputLabel>
            <Input type={props.type} value={props.value} onChange={e => props.handleOnChange(e.target.value)} aria-label={`${props.InputLabel}-input`}/>
        </FormControl>
    );
};

export default FormInput;