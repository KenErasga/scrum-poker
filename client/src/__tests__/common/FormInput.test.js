import React from 'react';
import renderer from 'react-test-renderer';
import FormInput from '../../common/FormInput';

describe('FromInput', () => {
    it('FormInput renders correctly', () => {
    const tree = renderer
        .create(<FormInput InputLabel='test-label' handleOnChange={(e) => e} type='test' value='value'/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
});
