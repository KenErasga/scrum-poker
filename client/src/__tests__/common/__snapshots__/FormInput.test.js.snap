// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromInput FormInput renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="component-simple"
  >
    test-label
  </label>
  <div
    aria-label="test-label-input"
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="test"
      value="value"
    />
  </div>
</div>
`;
